### PR TITLE
[1LP][RFR] service dialog regex validation

### PR DESCRIPTION
**Pull request by dgaikwad**
## Purpose or Intent
Adding test to check is validation regex text filter working or not

### PRT Run

{{pytest: cfme/tests/services/test_dialog_regex_validation_in_catalog.py::test_dialog_regex_validation_button cfme/tests/services/test_dialog_regex_validation_in_catalog.py::test_dialog_element_regex_validation --long-running -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit db450e2bd49145a3455134cb84a170f08dfd040d

The *local* fixture **`dialog_cat_item`** is used in the following files:
- **cfme/tests/services/test_dialog_regex_validation_in_catalog.py**
    - *test_dialog_element_regex_validation*
    - *test_dialog_regex_validation_button*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 1]
```diff
@@ -45,12 +34,23 @@ def dialog_cat_item(appliance, catalog):
...
+    'element_information': {
+        'ele_label': fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+        'ele_name': fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+        'ele_desc': fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+        'choose_type': "Text Box"
+    },
+    'options': {
+        'validation_switch': True,
+        'validation': "^([a-z0-9]+_*)*[a-z0-9]+$"
+    }}}], indirect=True)
```
Comment (john-dupuy) [inline cfme/tests/services/test_dialog_regex_validation_in_catalog.py:53]:
```suggestion
@pytest.mark.parametrize(
    "dialog_cat_item", [{"validation": "^([a-z0-9]+_*)*[a-z0-9]+$"}], indirect=True
)
```

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 2]
```diff
@@ -94,11 +95,30 @@ def test_dialog_text_area_element_regex_validation():
...
+                    "ele_label": fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+                    "ele_name": fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+                    "ele_desc": fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+                    "choose_type": "Text Box",
+                },
+                "options": {"validation_switch": True, "validation": "^[0-9]*$"},
+            }
+        }
+    ],
+    indirect=True,
```
Comment (john-dupuy) [inline cfme/tests/services/test_dialog_regex_validation_in_catalog.py:119]:
```suggestion
@pytest.mark.parametrize(
    "dialog_cat_item", [{"validation": "^[0-9]*$"}], indirect=True
)
```
```

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 43bc5617dae2d16fec7f2187a6211f915251100e

The *local* fixture **`dialog_cat_item`** is used in the following files:
- **cfme/tests/services/test_dialog_regex_validation_in_catalog.py**
    - *test_dialog_element_regex_validation*
    - *test_dialog_regex_validation_button*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_regex_validation_in_catalog.py [hunk 3]
```diff
@@ -94,11 +95,30 @@ def test_dialog_text_area_element_regex_validation():
     pass
 
 
-@pytest.mark.meta(coverage=[1720245])
-@pytest.mark.manual
+@pytest.mark.meta(automates=[1720245])
```
[the remark this answers is not in the document]
Comment (dgaikwad) [reply cfme/tests/services/test_dialog_regex_validation_in_catalog.py:102]: done

**Comment by dgaikwad** [comment]
@john-dupuy  All changes are done, please review PR

**Pull request merged**